src/Aspire.Hosting.Dapr/IDistributedApplicationBuilderExtensions.cs

```diff
@@ -26,4 +28,19 @@ public static IDistributedApplicationBuilder AddDapr(this IDistributedApplicatio
 
         return builder;
     }
+
+    public static IResourceBuilder<IDaprComponentResource> AddDaprComponent(this IDistributedApplicationBuilder builder, string name, string type, DaprComponentOptions? options = null)
+    {
+        var resource = new DaprComponentResource(name, type) { Options = options };
+
+        return builder
+            .AddResource(resource)
+            .WithAnnotation(new ManifestPublishingCallbackAnnotation(writer => WriteDaprComponentResourceToManifest(writer, resource)));
+    }
+
+    private static void WriteDaprComponentResourceToManifest(Utf8JsonWriter writer, DaprComponentResource resource)
+    {
+        writer.WriteString("type", "dapr.component.v0");
+        writer.WriteString("componentType", resource.Type);
```

Comment:
Is it enough to just reference the type here? On k8s are these component files uploaded to the cluster? Or is that a different installation process?

Reply:
That's a good question.  Since Dapr components are an abstraction, the specific implementations (and configuration) become a deployment concern rather than just part of the static application model (as typical with Aspire).  I'm not sure if there is an intent with Aspire for applications to alter their application model for a specific deployment target (e.g. how would that be specified, say, when doing an `azd deploy/up` which causes the manifest to be created).

 There are a couple of ways (or a mix of them) I can think of to manage this:

 - Component resources need not specify a specific type (just the building block). The deployment tool will then create an instance of an appropriate type of that building block (e.g. use the built-in Redis component for local development, but create Azure resources and appropriate Dapr configuration when deploying to Azure). The ability to do further customization or overrides would fall to the deployment tool (rather than be part of the application model).
 - Users use environment/configuration values to alter the application model based on a terget deployment target (e.g. local vs. Azure vs. Kubernetes). The application model creates specific target-specific resources (e.g. Azure Key Vault), which are then referenced by Dapr component resources.  The deployment tool then understands how to create/configure appropriate Dapr configuration (e.g. a "secretstore" Dapr component resource that references a Key Vault resource would end up generating "secretstore.keyvault" configuration wired up to connect to *that* specific service). 

The first option seems like it fits the Aspire philosophy of being highly opinionated and local-deployment-oriented.  At least, it seems like a good place to start.  It gets more complicated with Dapr input/output bindings, as they *are* a connection to an explicit resource, so I think we'd need to use the second option.  Still, we might be smart enough to map a referenced resource to a specific Dapr input/output binding.